### PR TITLE
Add "First Seen" Parser Flag

### PR DESCRIPTION
**Pull request by Maffooch**
Adds a "First Seen" parser flag to select parsers for users wanting the finding date to be reflected as the first time a vulnerability's detected rather then the last time

[sc-3965]

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

File: dojo/tools/veracode_sca/parser.py [hunk 1]
```diff
@@ -51,7 +52,15 @@ def _get_findings_json(self, file, test):
             if issue.get("issue_type") != "vulnerability":
                 continue
 
-            date = parser.parse(issue.get("created_date"))
+            # Get the date based on the first_seen setting
+            try:
+                if settings.USE_FIRST_SEEN:
+                    date = parser.parse(issue.get("created_date"))
+                else:
+                    date = parser.parse(issue.get("created_date"))
```
Comment (kiblik) [inline dojo/tools/veracode_sca/parser.py:60]: These looks same for me

File: dojo/tools/veracode_sca/parser.py [hunk 2]
```diff
@@ -166,9 +175,19 @@ def get_findings_csv(self, file, test):
...
-            )
+            # Get the date based on the first_seen setting
+            try:
+                if settings.USE_FIRST_SEEN:
+                    date = datetime.strptime(
+                        row.get("Issue opened: Scan date"), "%d %b %Y %H:%M%p %Z"
+                    )
+                else:
+                    date = datetime.strptime(
+                        row.get("Issue opened: Scan date"), "%d %b %Y %H:%M%p %Z"
```
Comment (kiblik) [inline dojo/tools/veracode_sca/parser.py:186]: These as well

File: dojo/tools/veracode_sca/parser.py (hunk 1 above)
Comment (Maffooch) [reply dojo/tools/veracode_sca/parser.py:60]: They are the same, but this time it is intentional. In the unit test reports for this parser, there is not any available data in the report to satisfy the notion of a "last seen" date. Before this change, the `created_date` field was used, so it seemed appropriate to maintain that same behavior for both cases.

I am open to any suggestions that could make more sense though!

File: dojo/tools/veracode_sca/parser.py (hunk 2 above)
Comment (Maffooch) [reply dojo/tools/veracode_sca/parser.py:186]: They are the same, but this time it is intentional. In the unit test reports for this parser, there is not any available data in the report to satisfy the notion of a "last seen" date. Before this change, the `Issue opened: Scan date` field was used, so it seemed appropriate to maintain that same behavior for both cases.

I am open to any suggestions that could make more sense though!

**Review by cneill — APPROVED** [review]

**Comment by kiblik** [comment]
One more parser ;) https://github.com/Maffooch/django-DefectDojo/pull/2

File: dojo/tools/veracode_sca/parser.py (hunk 1 above)
Comment (kiblik) [reply dojo/tools/veracode_sca/parser.py:60]: In that case, is it necessary to replace the original implementation if there is no change in behavior?

File: dojo/tools/veracode_sca/parser.py (hunk 2 above)
Comment (kiblik) [reply dojo/tools/veracode_sca/parser.py:186]: same as https://github.com/DefectDojo/django-DefectDojo/pull/9361/files#r1459718238

**Comment by Maffooch** [comment]
Thanks @kiblik

**Pull request merged**